code/modules/projectiles/ammo_datums.dm

```diff
@@ -2588,6 +2588,53 @@ datum/ammo/bullet/revolver/tp44
 	reagent_transfer_amount = 6.5
 
 
+/datum/ammo/xeno/corrosive
+	name = "Corrosive spit"
+	flags_ammo_behavior = AMMO_XENO|AMMO_EXPLOSIVE|AMMO_SKIPS_ALIENS
+	spit_cost = 200
+	added_spit_delay = 10
+	damage_type = BURN
+	accurate_range = 5
+	max_range = 10
+	accuracy_var_low = 3
+	accuracy_var_high = 3
+	damage = 10
+	smoke_strength = 0.9
+	smoke_range = 0.5
+	reagent_transfer_amount = 0.8
+
+/datum/ammo/xeno/corrosive/on_hit_mob(mob/living/carbon/C, obj/projectile/P)
+	drop_corrosive_smoke(get_turf(C))
+
+	if(!istype(C) || C.stat == DEAD || C.issamexenohive(P.firer) )
+		return
+
+	if(isnestedhost(C))
+		return
+
+/datum/ammo/xeno/corrosive/on_hit_obj(obj/O, obj/projectile/P)
+	var/turf/T = get_turf(O)
+	drop_corrosive_smoke(T.density ? P.loc : T)
+
+/datum/ammo/xeno/corrosive/on_hit_turf(turf/T, obj/projectile/P)
+	drop_corrosive_smoke(T.density ? P.loc : T)
+
+/datum/ammo/xeno/corrosive/do_at_max_range(turf/T, obj/projectile/P)
+	drop_corrosive_smoke(T.density ? P.loc : T)
+
+/datum/ammo/xeno/corrosive/set_smoke()
+	smoke_system = new /datum/effect_system/smoke_spread/xeno/corrosive()
+
+/datum/ammo/xeno/corrosive/proc/drop_corrosive_smoke(turf/T)
```

Comment:
I know this is how the other spits have done it, but there's really no reason for these separate procs, it should all just be one drop_smoke proc, or really you could just use drop_nade.